multistate_kernel/util.py

```diff
@@ -104,6 +104,28 @@ def _x_2d_from_1d(x_1d_):
             [np.full_like(x, i).reshape(-1,1), np.asarray(x).reshape(-1,1)] for i, x in enumerate(x_1d_)
         ]).reshape(-1, 2)
 
+    def __mul__(self, other):
+        norm = self.norm * other
+        return MultiStateData.from_arrays(self.arrays.x, self.arrays.y, self.arrays.err, norm, keys=self.keys())
+
+    def __rmul__(self, other):
+        return self * other
+
+    def __add__(self, other):
+        if not np.allclose(self.arrays.x, other.arrays.x):
+            raise ValueError('objects should have the same x values')
+        if set(self.keys()) - set(other.keys()):
+            raise ValueError('objects should have the same keys()')
+        x = self.arrays.x
+        y = self.arrays.y * self.arrays.norm + other.arrays.y * other.arrays.norm
+        norm = self.__norm(y)
+        y /= norm
+        err = np.sqrt((self.arrays.err * self.arrays.norm)**2 + (other.arrays.err * other.arrays.norm)**2) / norm
+        return MultiStateData.from_arrays(x, y, err, norm, keys=self.keys())
+
+    def __sub__(self, other):
+        return self + (-1) * other
```

Comment:
Well, it is so pythonic to implement subtraction using one extra object construction.